### ISSUE TITLE
Type in container cannot be inferred from unbound method

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#484]
**Describe the bug**
When calling an instance method not bound to an object on an object of the same type, the result is assigned to a container of unknown type items.

**To Reproduce**
Type check the snippet below and observe that the revealed type is `KeysView[Unknown]`.

**Expected behavior**
That the revealed type would be `KeysView[str]`.

**Screenshots or Code**
```py
from typing import Dict

foo: Dict[str, str] = {}
reveal_type(dict.keys(foo))
```

**VS Code extension or command-line**
extension, 1.1.19

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is now fixed in 1.1.20, which I just published.
